includes/drush.inc

```diff
@@ -1134,6 +1135,32 @@ function drush_preflight_command_dispatch() {
     drush_set_context('cli', $cli_options);
     _drush_preflight_global_options();
   }
+  $args = drush_get_arguments();
+  $command_name = array_shift($args);
+  $root = drush_get_context('DRUSH_SELECTED_DRUPAL_ROOT');
+  $local_drush = drush_get_option('drush-script');
+  $is_local = drush_get_option('local');
+  $values = NULL;
+  if (!empty($root) && !empty($local_drush) && empty($is_local)) {
+    if (!drush_is_absolute_path($local_drush)) {
+      $local_drush = $root . '/' . $local_drush;
+    }
+    $local_drush = realpath($local_drush);
+    $this_drush = realpath($GLOBALS['argv'][0]);
```

Comment:
Do we really want to grab from GLOBALS here? Maybe drush_build_drush_command() is helpful?